server/service/client.go

```diff
@@ -468,6 +468,25 @@ func (c *Client) ApplyGroup(
 	}
 
 	if specs.AppConfig != nil {
+		// Clear out SSO settings if not provided.
+		if _, ok := specs.AppConfig.(map[string]interface{})["sso_settings"]; !ok {
+			specs.AppConfig.(map[string]interface{})["sso_settings"] = fleet.SSOSettings{}
+		}
+
+		// Clear out SMTP settings if not provided.
+		if _, ok := specs.AppConfig.(map[string]interface{})["smtp_settings"]; !ok {
+			specs.AppConfig.(map[string]interface{})["smtp_settings"] = fleet.SMTPSettings{}
+		}
+
+		// Clear out MDM end_user_authentication settings if not provided.
+		// TODO -- clear out all MDM settings if not provided?
+		if _, ok := specs.AppConfig.(map[string]interface{})["mdm"]; ok {
+			mdm := specs.AppConfig.(map[string]interface{})["mdm"]
+			if _, ok := mdm.(map[string]interface{})["end_user_authentication"]; !ok {
+				mdm.(map[string]interface{})["end_user_authentication"] = fleet.MDMEndUserAuthentication{}
+			}
+		}
```

Comment:
I'm being deliberately targeted here.  Technically we should clear out everything from MDM if the `mdm` key is left out, but we should investigate just which things are and are not being cleared right now, and possibly have a bigger warning if we change that.